app/assets/javascripts/alchemy/alchemy.link_dialog.js.coffee

```diff
@@ -45,44 +46,71 @@ class window.Alchemy.LinkDialog extends Alchemy.Dialog
         when 'anchor'
           url = @$anchor_link.val()
         else
-          url = $("##{@link_type}_urlname").val()
+          url = @$page_urlname.val()
+          if @$element_anchor.val() != ''
+            url += "##{@$element_anchor.val()}"
       # Create the link
       @createLink
         url: url
         title: $("##{@link_type}_link_title").val()
         target: $("##{@link_type}_link_target").val()
       false
 
-  # Attaches click events to buttons in the link dialog that appear after the
-  # page tree has finished loading.
-  attachTreeEvents: ->
-    # The select page and show elements links.
-    $('a.sitemap_pagename_link, a.show_elements_to_link', @dialog_body).click (e) =>
-      $this = $(e.currentTarget)
-      page_id = $this.data('page-id')
-      url = $this.data('url')
-      # Select page in page tree
-      @selectPage(page_id)
-      # if the show elements link was clicked
-      if $this.hasClass('show_elements_to_link')
-        # we open the elements select for that page
-        @showElementsSelect($this.attr('href'), url)
-      else
-        # store url
-        @$internal_urlname.val('/' + url)
-      false
-    # Select the current page in the tree
-    @initInternalLinkTab()
-
-  # Sets the page selected and scrolls it in the viewport.
-  selectPage: (page_id) ->
-    # deselect any selected page from page tree
-    @deselectPage()
-    $('#sitemap_sitename_' + page_id).addClass('selected_page')
-    @$page_container.scrollTo("#sitemap_sitename_#{page_id}", {duration: 400, offset: -10})
-
-  deselectPage: ->
-    $('#sitemap .selected_page', @dialog_body).removeClass('selected_page')
+  # Initializes the select2 based Page select
+  initPageSelect: ->
```

Comment:
Hm just for your sanity: Maybe write new features in JS instead? :)

Reply:
And convert the existing file from Coffeescript to Javascript before making that change? That way the diff/change of this PR would be much higher. This is a new feature, not a new file.